hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -589,12 +568,26 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		})
 	}
 
-	// Validate config input.
-	// 3 generic core config resources: fips, ssh and haproxy.
-	// TODO (alberto): consider moving the expectedCoreConfigResources check
-	// into the token Secret controller so we don't block Machine infra creation on this.
-	expectedCoreConfigResources := expectedCoreConfigResourcesForHostedCluster(hcluster)
-	config, mirroredConfigs, missingConfigs, err := r.getConfig(ctx, nodePool, expectedCoreConfigResources, controlPlaneNamespace, releaseImage, hcluster)
+	// TODO(alberto): move this into a validation section.
+	// Retrieve pull secret name to check for changes when config is checked for updates
+	_, err = r.getPullSecretName(ctx, hcluster)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	additionalTrustBundleCM := &corev1.ConfigMap{}
+	if hcluster.Spec.AdditionalTrustBundle != nil {
+		additionalTrustBundleCM, err = r.getAdditionalTrustBundle(ctx, hcluster)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	haproxyRawConfig, err := r.generateHAProxyRawConfig(ctx, hcluster, releaseImage)
```

Comment:
we could call `generateHAProxyRawConfig` function directly in `NewConfigGenerator` instead of passing it as input.

Reply:
I think the `ConfigGenerator` isn't meant to be aware of reconciliation and how to generate the HAProxyRawConfig. Declaring its input/dependecies as parameters in the constructor function avoid necessary tight-coupling with the reconciler and makes it easier to test. Just my high level observation from my limited familiarity with the code.